tests/lib/rules/jest/no-vague-titles.js

```diff
@@ -0,0 +1,390 @@
+const {RuleTester} = require('eslint');
+const rule = require('../../../../lib/rules/jest/no-vague-titles');
+
+const ruleTester = new RuleTester();
+
+require('babel-eslint');
+
+const parser = 'babel-eslint';
+
+function errorWithMethod(method, type = 'CallExpression') {
+  return [
+    {
+      type,
+      message: `${method} statement should not contain vague words`,
+    },
+  ];
+}
+ruleTester.run('no-vague-titles', rule, {
+  valid: [
+    {
+      code: `it('foo bar baz')`,
+      parser,
+    },
+    {
+      code: `describe('foo bar baz')`,
+      parser,
+    },
+    {
+      code: `test('foo bar baz')`,
+      parser,
+    },
+  ],
+  invalid: [
+    {
+      code: "it('correct')",
+      parser,
+      errors: errorWithMethod('it'),
+    },
+    {
+      code: "it('correctly')",
+      parser,
+      errors: errorWithMethod('it'),
+    },
+    {
+      code: "it('incorrect')",
+      parser,
+      errors: errorWithMethod('it'),
+    },
+    {
+      code: "it('Correct')",
+      parser,
+      errors: errorWithMethod('it'),
+    },
+    {
+      code: "it.only('correct')",
+      parser,
+      errors: errorWithMethod('it', 'MemberExpression'),
+    },
+    {
+      code: "it.only('correctly')",
+      parser,
+      errors: errorWithMethod('it', 'MemberExpression'),
+    },
+    {
+      code: "it.only('incorrect')",
+      parser,
+      errors: errorWithMethod('it', 'MemberExpression'),
+    },
+    {
+      code: "it.only('Correct')",
+      parser,
+      errors: errorWithMethod('it', 'MemberExpression'),
+    },
+    {
+      code: "describe('correct')",
+      parser,
+      errors: errorWithMethod('describe'),
+    },
+    {
+      code: "describe('correct')",
```

Comment:
I would ignore for now, but you might be interested in tackling the new `describe/it/test.each` Jest recently introduced: https://facebook.github.io/jest/docs/en/api.html#describeeachtable-name-fn